packages/shared/components/popups/SendConfirmationPopup.svelte

```diff
@@ -54,17 +56,37 @@
     $: $$props, expirationDate, _prepareOutput()
 
     function onConfirm(): void {
-        closePopup()
-
         if ($isSoftwareProfile) {
             void send()
         } else if ($isLedgerProfile) {
-            promptUserToConnectLedger(false, () => send(), undefined)
+            closePopup()
+            promptUserToConnectLedger(false, () => send(false), undefined)
         }
     }
 
-    function send(): Promise<void> {
-        return trySendOutput(outputOptions, preparedOutput)
+    function send(shouldClosePopup: boolean = true): Promise<void> {
+        function _send(): Promise<void> {
+            if (shouldClosePopup) {
+                closePopup()
+            }
+            return trySendOutput(outputOptions, preparedOutput)
+        }
+
+        if (expirationDate) {
+            if (isValidExpirationDateTime(expirationDate)) {
```

Comment:
This can go into one condition

Reply:
Or it can be removed since it is checked inside isValidExpirationDateTime as well.

Reply:
Never mind, I needed to read this a bit more carefully, I left the logic as is.